### PR TITLE
enabling mermaid diagrams and adding sequence diagram for branching workflows

### LINKED ISSUE #77: to add a worfklow picture regarding the branching workflows
in order to help in a better understanding on how to collaborate within the opendevstack project we could add a picture to enhance the written description of the branching workflows.

### PR DESCRIPTION
**Pull request by gerardcl**
proposal for a contributing flow diagram in the development guide documentation
closes #77 

one can test the sequence diagram provided at https://mermaidjs.github.io/mermaid-live-editor/ 
by pasting in the code input (same code as in development-guide.md):
```
sequenceDiagram
    participant A as OpenDevStack <br />GitHub
    participant B as ACME GitHub <br />OpenDevStack fork
    participant C as ACME private repo <br />OpenDevStack fork
    A->>C: master branch checkout
    C->>C: new branch X from master branch
    C->>C: validate branch X
    C-->>C: PR: branch X to production branch
    C->>C: validate production branch change
    C->>B: push branch X
    B-->>A: PR: branch X to master
```

### CODE REVIEW

**Comment by michaelsauter** [comment]
@gerardcl Thanks! Two notes:

* I would remove the "validate" arrows, they were a bit confusing for me first. Maybe there is another way to explain that step?
* I'd say using `master` as base is rather unusual - maybe better would be to use a branch (e.g. `1.0.x`) or a tag (`v1.0.1`) as an example ...

**Comment by gerardcl** [comment]
Hi @michaelsauter,
what about this? see the new commit too.

sequenceDiagram
    participant A as OpenDevStack <br />GitHub
    participant B as ACME GitHub <br />OpenDevStack fork
    participant C as ACME private repo <br />OpenDevStack fork
    A->>C: 1.0.x/v1.0.x checkout
    C->>C: new branch X from 1.0.x/v1.0.x branch
    C->>C: develop and test the new feature branch X
    C-->>C: PR: branch X to production branch
    C->>C: test production branch changes
    C->>B: push branch X
    B-->>A: PR: branch X to 1.0.x/v1.0.x

File: doc/development-guide.md [hunk 1]
```diff
@@ -104,3 +104,22 @@ ensure the following:
...
+
+Next sequence diagram provides a graphical understanding on the steps to follow in order to add a new functionality to the OpenDevStack official repository from ACME's private fork.
+The example is based on the assumption you want to contribute with a new feature from branch *1.0.x* or tag *v1.0.x*
+
+<div class="mermaid">
+sequenceDiagram
+    participant A as OpenDevStack <br />GitHub
+    participant B as ACME GitHub <br />OpenDevStack fork
+    participant C as ACME private repo <br />OpenDevStack fork
+    A->>C: 1.0.x/v1.0.x checkout
```
Comment (michaelsauter) [inline doc/development-guide.md]: I think this should be `A->>C: 1.0.x/v1.0.1 checkout` to show a concrete example.

File: doc/development-guide.md [hunk 2]
```diff
@@ -104,3 +104,22 @@ ensure the following:
...
+    C->>C: new branch X from 1.0.x/v1.0.x branch
+    C->>C: develop and test the new feature branch X
```
Comment (michaelsauter) [inline doc/development-guide.md]: I would do the above two lines as comments.

File: doc/development-guide.md [hunk 3]
```diff
@@ -104,3 +104,22 @@ ensure the following:
...
+    C-->>C: PR: branch X to production branch
+    C->>C: test production branch changes
```
Comment (michaelsauter) [inline doc/development-guide.md]: I would do the above line as comment.

File: doc/development-guide.md (hunk 1 above)
Comment (gerardcl) [reply doc/development-guide.md]: done

File: doc/development-guide.md (hunk 2 above)
Comment (gerardcl) [reply doc/development-guide.md]: see proposal

File: doc/development-guide.md (hunk 3 above)
Comment (gerardcl) [reply doc/development-guide.md]: done

**Review by michaelsauter — APPROVED** [review]

**Pull request merged**